packages/doenetml-worker-rust/lib-doenetml-core/src/core/state/types/math_expr.rs

```diff
@@ -0,0 +1,429 @@
+use std::collections::HashMap;
+use strum_macros::Display;
+
+use crate::math_via_wasm::{
+    eval_js, math_to_latex, normalize_math, parse_latex_into_math, parse_text_into_math,
+    substitute_into_math,
+};
+
+const BLANK_MATH_OBJECT: &str = "{\"objectType\":\"math-expression\",\"tree\":\"\u{ff3f}\"}";
+/// A mathematical expression represented as a serialized `math-expressions` object.
+#[derive(Debug, Clone, PartialEq, serde::Deserialize)]
+pub struct MathExpr {
+    pub math_object: String,
+}
+
+impl Default for MathExpr {
+    fn default() -> Self {
+        MathExpr {
+            math_object: BLANK_MATH_OBJECT.to_string(),
+        }
+    }
+}
+
+impl std::fmt::Display for MathExpr {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.to_latex(ToLatexParams::default()))
+    }
+}
+
+impl serde::Serialize for MathExpr {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(&self.math_object)
+    }
+}
+
+impl MathExpr {
+    /// Create a `Math` by parsing the string `text` using the `math-expressions` text parser.
+    ///
+    /// Parameters:
+    /// - `text`: the string to be parsed
+    /// - `split_symbols`: if true, split multi-character symbols that do not contain an number
```

Comment:
Finally, a description :-D. I think `single_char_symbols` would be a better name.

Reply:
Maybe, but `splitSymbols` is the current DoenetML word used for the attribute. We could have the source code deviate from the DoenetML term if there is a strong enough reason, but I try to minimize that. I've generally gotten resistance at making names longer, so I'm not sure if `singleCharSymbols` would be approved for DoenetML.

Reply:
I didn't know it was a `DoenetML` keyword.